OK.

-Wzero-as-null-pointer-constant
```
[  5%] Building CXX object CMakeFiles/gmodel.dir/gmodel.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++     --std=c++11 -O2 -g -fno-omit-frame-pointer -ferror-limit=1 -Werror -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-padded -Wno-float-equal -Wno-weak-template-vtables -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk   -o CMakeFiles/gmodel.dir/gmodel.cpp.o -c /Users/cwsmith/develop/gmodel/gmodel.cpp
/Users/cwsmith/develop/gmodel/gmodel.cpp:33:21: error: zero as null pointer constant [-Werror,-Wzero-as-null-pointer-constant]
    /* LOOP    = */ 0,
                    ^
                    nullptr
fatal error: too many errors emitted, stopping now [-ferror-limit=]
2 errors generated.
```

-Wunused-template
```
[  5%] Building CXX object CMakeFiles/gmodel.dir/gmodel.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   --std=c++11 -O2 -g -fno-omit-frame-pointer -ferror-limit=1 -Werror -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-padded -Wno-float-equal -Wno-weak-template-vtables-Wno-zero-as-null-pointer-constant -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk   -o CMakeFiles/gmodel.dir/gmodel.cpp.o -c /Users/cwsmith/develop/gmodel/gmodel.cpp
/Users/cwsmith/develop/gmodel/gmodel.cpp:67:12: error: unused function template 'size'
      [-Werror,-Wunused-template]
static int size(std::vector<T> const& v) {
           ^
1 error generated.
```

--- a/gmodel.cpp
+++ b/gmodel.cpp
@@ -447,7 +447,7 @@ Extruded extrude_edge3(ObjPtr start, Transform tr, Extruded left, Extruded right
   auto loop = new_loop();
   add_use(loop, FORWARD, start);
   add_use(loop, FORWARD, right.middle);
-  ObjPtr end = 0;
+  ObjPtr end = nullptr;
   switch (start->type) {
     case LINE: {
       end = new_line2(std::dynamic_pointer_cast<Point>(left.end),
@@ -487,7 +487,7 @@ Extruded extrude_edge3(ObjPtr start, Transform tr, Extruded left, Extruded right
       break;
     }
     default:
-      end = 0;
+      end = nullptr;
       break;
   }
   add_use(loop, REVERSE, end);
@@ -503,7 +503,7 @@ Extruded extrude_edge3(ObjPtr start, Transform tr, Extruded left, Extruded right
       middle = new_ruled2(loop);
       break;
     default:
-      middle = 0;
+      middle = nullptr;
       break;
   }
   return Extruded{middle, end};

